enhancements/test-platform/e2e-observer-pods.md

```diff
@@ -0,0 +1,98 @@
+ 4. loki as provided by the logging team (future)
+To debug our existing install and upgrade failures.
+
+### Goals
+
+ 1. allow a cluster observing tool to run in the CI cluster (this avoid restarts during upgrades)
+ 2. allow a cluster observer to provide data (including junit) to be collected by the CI job
+
+### Non-Goals
+
+ 1. allow a cluster observer to impact success or failure of a job
+ 2. allow a cluster observer to impact the running test.  This is an observer author failure.
+ 3. provide ANY dimension specific data. If an observer needs this, they need to integrate differently.
+
+## Proposal
+
+### Changes to CI
+This is a sketch of a possible path.
+ 1. Allow a non-dptp-developer to produces a pod template that mounts a `secret/<job>-kubeconfig` that will later contain a kubeconfig.
+ 2. When the CI pod starts today (the one with setup containers and stuff), also create an instance of each pod template,
+    let's say `pod/<job>-resourcewatcher` and an empty `secret/<job>-kubeconfig`.
+ 3. As soon as a kubeconfig is available (these go into a known location in setup container today), write that
+    `kubeconfig` into every `secret/<job>-kubeconfig` (you probably want to label them).
+ 4. When it is time for collection, the existing pod (I think it's teardown container), writes a new data entry at `.data['teardown']` into every 
+    `secret/<job>-kubeconfig`.
+    The value should be a timestamp.
+ 5. Ten minutes after `.data['teardown']` is written, the teardown container rsyncs a well known directory,
+    `/var/e2e-observer`, which may contain `/var/e2e-observer/junit` and `/var/e2e-observer/artifacts`.  These contents
+    are placed in some reasonable spot.
+    
+    This could be optimized with a file write in the other direction, but the naive approach doesn't require it.
+  6. All `pod/<job>-<e2e-observer-name>` are deleted.
+
+### Requirements on e2e-observer authors
+ 1. Your pod must be able to run against *every* dimension. No exceptions.  If your pod needs to quietly no-op, it can do that.
+ 2. Your pod must be able to tolerate a kubeconfig that doesn't work.  The kubeconfig may point to a cluster than never comes up.
+    Or that hasn't come up yet.  Your pod must not suck on it.
+ 3. If your pod fails, it will not fail the e2e job.  If you need to fail an e2e job reliably, you need something else.
+ 4. Your pod must terminate when asked.
+
+## Alternatives
+
+### Modify each type of job
+Given the matrix of dimensions, this seems impractical.
+Even today, we have a wide variance in the quality of artifacts from different jobs.
+
+### Modify the openshift-tests command
+This is easy for some developers, BUT as e2e monitor shows us, it doesn't provide enough information.
+We need information from before the test command is run to find many of our problems.
+Missing logs and missing intermediate resource states (intermediate operator status) are the most egregious so far. 
```

Comment:
The vanilla e2e suite already records intermediate operator status in `Monitor cluster while tests execute`.  It does not seem like that big a reach to me to ask the installer to also monitor and log operator status, instead of its current "just collect this at the end if the install failed" behavior.  And that would also be useful information to have in the log for installs that fail in the wild, which will not benefit from CI-specific monitoring suites.